src/services/urlSourcedFetcher.ts

```diff
@@ -76,6 +76,14 @@ export class UrlSourcedFetcher {
       models = JSON.parse(content);
     } else if (this.mediaType.startsWith("text/csv")) {
       models = await csv({ flatKeys: true }).fromString(content);
+    } else if (this.mediaType.startsWith("application/vnd.ms-excel") && this.file.name.endsWith(".csv")) {
+      // Windows can sometimes report the file type of a csv as an
+      // application/vnd.ms-excel file type.
+      // Therefore, if the file type is reported as "excel" and the file
+      // extension is .csv, it is safe to assume that Windows is incorrectly
```

Comment:
I wonder if this is less about correctness and more to do with the default application registered to open csv files. Actually I'd bet it is this, and FF uses the Windows mine database or something

Reply:
I can replicate the bug on my computer and the default application I have for csv files is notepad.exe

Reply:
Updated comment

![image](https://github.com/user-attachments/assets/1ccd6f85-d5c9-4dcc-ae4a-b76e0fc72f28)